File: frontend/src/microservices/attraction/objects/CreateAttractionTicketRulePlannerRequest.ts
// 本文件定义 attraction 模块的 `CreateAttractionTicketRulePlannerRequest`，作为planner 请求参数并提供 JSON 编解码。

export type CreateAttractionTicketRulePlannerRequest = {
  managerId: string
  attractionId: string
  ticketTypeId: string
  ruleType: string
  ageValue?: number | null
  minAge?: number | null
  maxAge?: number | null
  documentType?: string | null
  documentNumberPrefix?: string | null
}

export const createAttractionTicketRulePlannerRequestFromJson = (json: string): CreateAttractionTicketRulePlannerRequest =>
  JSON.parse(json) as CreateAttractionTicketRulePlannerRequest

export const createAttractionTicketRulePlannerRequestToJson = (value: CreateAttractionTicketRulePlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/attraction/objects/CreateAttractionTicketRulePlannerRequest.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateAttractionTicketRulePlannerRequest(
    managerId: String,
    attractionId: String,
    ticketTypeId: String,
    ruleType: String,
    ageValue: Option[Int],
    minAge: Option[Int],
    maxAge: Option[Int],
    documentType: Option[String],
    documentNumberPrefix: Option[String]
)
object CreateAttractionTicketRulePlannerRequest:
  given sourceEncoder: Encoder[CreateAttractionTicketRulePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateAttractionTicketRulePlannerRequest] = deriveDecoder

